### PR TITLE
Dark mode preference

### PR DESCRIPTION
**Pull request by zahin-mohammad**
- Implement custom toolbar
- Implement preference
- Change ui mode based on preference

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -51,6 +51,8 @@ dependencies {
     implementation "com.google.android.material:material:1.2.0-alpha02"
     implementation "androidx.constraintlayout:constraintlayout:1.1.3"
     implementation 'com.github.PhilJay:MPAndroidChart:v3.1.0'
+    implementation "com.android.support:preference-v7:28.0.0"
+    implementation "com.android.support:appcompat-v7:28.0.0"
```
Comment (brian-norman) [inline app/build.gradle]: It seems you've used the deprecated pre-androidx preferences library, can you update to the jetpack one?
https://developer.android.com/jetpack/androidx/releases/preference

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
         <item name="colorAccent">@color/colorAccent</item>
+
+    </style>
+
+    <style name="ToolBarStyle" parent="Theme.AppCompat">
+        <item name="android:textColorPrimary">@android:color/white</item>
+        <item name="android:textColorSecondary">@android:color/white</item>
+        <item name="actionMenuTextColor">@android:color/white</item>
+<!--        <item name="actionOverflowButtonStyle">@style/ActionButtonOverflowStyle</item>-->
+<!--        <item name="drawerArrowStyle">@style/DrawerArrowStyle</item>-->
```
Comment (brian-norman) [inline app/src/main/res/values/styles.xml]: 😕

File: app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
         <item name="colorPrimary">@color/colorPrimary</item>
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
+
+    </style>
+
+    <style name="ToolBarStyle" parent="Theme.AppCompat">
+        <item name="android:textColorPrimary">@android:color/white</item>
+        <item name="android:textColorSecondary">@android:color/white</item>
+        <item name="actionMenuTextColor">@android:color/white</item>
```
Comment (brian-norman) [inline app/src/main/res/values/styles.xml:15]: Cool you solved it 😄

**Review by brian-norman — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (brian-norman) [reply app/build.gradle]: `implementation "androidx.preference:preference-ktx:$preference_version"`

Comment (zahin-mohammad) [reply app/build.gradle]: Nice catch, will update.

File: app/src/main/res/values/styles.xml (hunk 3 above)
Comment (zahin-mohammad) [reply app/src/main/res/values/styles.xml]: 😮

File: app/src/main/res/values/styles.xml (hunk 2 above)
Comment (zahin-mohammad) [reply app/src/main/res/values/styles.xml]: 😮

File: app/build.gradle (hunk 1 above)
Comment (zahin-mohammad) [reply app/build.gradle]: Updated.

File: app/src/main/res/values/styles.xml (hunk 2 above)
Comment (zahin-mohammad) [reply app/src/main/res/values/styles.xml]: Removed.

**Review by brian-norman — APPROVED** [review]

**Pull request merged**